build/main_test.go

```diff
@@ -34,3 +34,45 @@ func ExampleTransactionBuilder() {
 	fmt.Printf("tx base64: %s", txeB64)
 	// Output: tx base64: AAAAADZY/nWY0gx6beMpf4S8Ur0qHsjA8fbFtBzBx1cbQzHwAAAAZAAAAAAAAAABAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAALSRpLtCLv2eboZlEiHDSGR6Hb+zZL92fbSdNpObeE0EAAAAAAAAAAB3NZQAAAAAAAAAAARtDMfAAAABA2oIeQxoJl53RMRWFeLB865zcky39f2gf2PmUubCuJYccEePRSrTC8QQrMOgGwD8a6oe8dgltvezdDsmmXBPyBw==
 }
+
+// ExamplePathPayment creates and signs a simple transaction with PathPayment operation, and then
+// encodes it into a base64 string capable of being submitted to stellar-core.
+func ExamplePathPayment() {
```

Comment:
Notice that I reuse the normal "Amount" mutators to provide the source asset and the source max.  It's a bit awkward with regards to communicating the fact that the source amount is a maximum, but IMO the consistency of use is important.  If we were to go your route using the `Asset` as a separate struct, IMO we should do the same thing with the non-path payment.  That is to say `NativeAmount` and `CreditAmount` would be replaced with 3 mutators:  `NativeAsset{}`, `CreditAsset{Code,Issuer}` and `Amount{string}`.